graphqlapi/src/main/java/io/stargate/graphql/graphqlservlet/HTTPAwareContextImpl.java

```diff
@@ -97,4 +112,58 @@ private boolean hasJWTToken() {
   public Optional<DataLoaderRegistry> getDataLoaderRegistry() {
     return Optional.ofNullable(dataLoaderRegistry);
   }
+
+  public BatchContext getBatchContext() {
+    return batchContext;
+  }
+
+  /**
+   * Encapsulates logic to add multiple statements contained in the same operation that need to be
+   * executed in a batch.
+   */
+  public static class BatchContext {
+    private final List<String> statements = Collections.synchronizedList(new ArrayList<>());
+    private final AtomicInteger statementCounter = new AtomicInteger();
+    private final CompletableFuture<ResultSet> future = new CompletableFuture<>();
+    private volatile DataStore dataStore;
+    private final AtomicInteger dataStoreCounter = new AtomicInteger();
+
+    public CompletableFuture<ResultSet> getExecutionFuture() {
+      return future;
+    }
+
+    public List<String> getStatements() {
+      return statements;
+    }
+
+    public void setExecutionResult(CompletableFuture<ResultSet> result) {
+      result.whenComplete(
+          (r, e) -> {
+            if (e != null) {
+              future.completeExceptionally(e);
+            } else {
+              future.complete(r);
+            }
+          });
+    }
+
+    public void setExecutionResult(Exception ex) {
+      future.completeExceptionally(ex);
+    }
+
+    public int add(String query) {
+      statements.add(query);
+      return statementCounter.incrementAndGet();
```

Comment:
Why we cannot just return `statements.size()`? I think that we don't need to have the `statementCounter`

Reply:
It's a synchronized list, so each call will require a lock acquisition.

I think I'll change it to a synchronized block instead.